Kudu.Core/Deployment/Generator/AspNetCoreBuilder.cs

```diff
@@ -6,25 +6,23 @@ namespace Kudu.Core.Deployment.Generator
 {
     class AspNetCoreBuilder : GeneratorSiteBuilder
     {
-        private readonly string _projectPath;
+        private readonly string _projectFilePath;
         private readonly string _solutionPath;
         private readonly string _version;
 
-        public AspNetCoreBuilder(IEnvironment environment, IDeploymentSettingsManager settings, IBuildPropertyProvider propertyProvider, string sourcePath, string projectPath, string solutionPath = null)
+        public AspNetCoreBuilder(IEnvironment environment, IDeploymentSettingsManager settings, IBuildPropertyProvider propertyProvider, string sourcePath, string projectFilePath, string solutionPath = null)
             : base(environment, settings, propertyProvider, sourcePath)
         {
-            _projectPath = projectPath; // either xproj, csproj or project.json
+            _projectFilePath = projectFilePath; // either xproj, csproj or project.json
             _solutionPath = solutionPath;
-            if (_projectPath.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase))
+            if (_projectFilePath.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase))
             {
                 _version = "csproj";
             }
-            else if (_projectPath.EndsWith(".xproj", StringComparison.OrdinalIgnoreCase))
+            else if (_projectFilePath.EndsWith(".xproj", StringComparison.OrdinalIgnoreCase))
             {
                 // if it's xproj, throw invalidOperationException
-                throw new InvalidOperationException(String.Format(CultureInfo.CurrentCulture,
-                                                             Resources.Error_ProjectNotDeployable,
-                                                             projectPath));
+                throw new InvalidOperationException("Building Asp.Net Core .xproj is no longer supported in Azure, please move to .csproj");
```

Comment:
Also add this link I just created for it in the error message: https://go.microsoft.com/fwlink/?linkid=850964